### PR TITLE
[Rgen] Add the BindingData struct as part as the code struct.

### PR DESCRIPTION
**Pull request by mandel-macaque**
This addition allows the code emitters to access the binding type configuration.

### CODE REVIEW

File: src/rgen/Microsoft.Macios.Generator/AttributesNames.cs [hunk 1]
```diff
@@ -15,4 +20,19 @@ public static class AttributesNames {
...
+			return BindingCategoryAttribute;
+		}
+		if (type == typeof (ObjCBindings.Class)) {
+			return BindingClassAttribute;
+		}
+		if (type == typeof (ObjCBindings.Protocol)) {
+			return BindingProtocolAttribute;
+		}
+
+		return null;
```
Comment (dalexsoto) [inline src/rgen/Microsoft.Macios.Generator/AttributesNames.cs:36]: Not entirely sure which one I prefer tbh (I lean just a tiny bit on to the switch but not for much), you make the call, I am fine either way.

```suggestion
		var type = typeof (T);
		switch (type) {
			case Type _ when type == typeof (ObjCBindings.Field):
				return ExportFieldAttribute;
			case Type _ when type == typeof (ObjCBindings.Property):
				return ExportPropertyAttribute;
			case Type _ when type == typeof (ObjCBindings.Method):
				return ExportMethodAttribute;
			default:
				return null;
		}
```

**Review by dalexsoto — APPROVED** [review]

File: src/rgen/Microsoft.Macios.Generator/AttributesNames.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/rgen/Microsoft.Macios.Generator/AttributesNames.cs:36]: the one with the switch to be looks very verbose and a switch become an if at the end of the day with a little better branching: https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpHqHFqZwAMA/OgCoCmAzgC7oAUAlgHZvcCuAWwBGDAE4BKfKWLFOAMw4CR49AF416SlIJJZ+sgHZ0AIgBe4gPYmA3DP0Bfe7OdzF7ZaLHrNcHa4MoYxNLbgZbAPQnPQMSGMDjAQAbJLt4qPtncigqWkZWOA4ePiEvf3TZJgB3ThYAYwALJVLxctj9OoBDJgYtEEj2oNMLMWs09tkunvQ4foqJoiGQsIj5g2AGeU7+JJY5hYT0ZNTI6PtohyA=

I don't know if it is worth the change.

Comment (dalexsoto) [reply src/rgen/Microsoft.Macios.Generator/AttributesNames.cs:37]: I am fine either way so not really pushing for a change 👍

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**